packages/dd-trace/src/plugins/util/user-provided-git.js

```diff
@@ -27,10 +27,11 @@ function removeEmptyValues (tags) {
   }, {})
 }
 
-// The regex is extracted from
+// The regex is inspired by
 // https://github.com/jonschlinkert/is-git-url/blob/396965ffabf2f46656c8af4c47bef1d69f09292e/index.js#L9C15-L9C87
+// The `.git` suffix is optional in this version
 function validateGitRepositoryUrl (repoUrl) {
-  return /(?:git|ssh|https?|git@[-\w.]+):(\/\/)?(.*?)(\.git)(\/?|#[-\d\w._]+?)$/.test(repoUrl)
+  return /(?:git|ssh|https?|git@[-\w.]+):(\/\/)?(.*?)(\.git)?(\/?|#[-\d\w._]+?)$/.test(repoUrl)
```

Comment:
It is not a current change, but why this `(.*?)`? it sounds weird, `*` makes the content optional, why add also the `?`?

Reply:
I think this grabs whatever possible before finding `.git`. Now that `.git` is optional this might not be the best way to do this

Reply:
I removed it